### PR TITLE
[INFINITY-2096][INFINITY-2109] Wait for recovery plan to start before checking success

### PR DESCRIPTION
**Pull request by elezar**
This PR attempts to address the flakiness of the Cassandra recovery tests (see https://jira.mesosphere.com/browse/INFINITY-2096). It was noticed that the default state of the recovery plan is `COMPLETE` which lead to the wrong things being checked at the various stages of the integration tests.

In addition, checking for `IN_PROGRESS` is not a reliable way to check if a deployment plan was started, and as such support for multiple conditions was added to the `sdk_plan.wait_for_plan_status` function.

### CODE REVIEW

File: testing/sdk_install.py [hunk 1]
```diff
@@ -12,15 +12,30 @@
 import sdk_plan
 import sdk_utils
 
+from retrying import retry
 
-def install(
+
+@retry(stop_max_attempt_number=3, retry_on_exception=lambda e: isinstance(e, dcos.errors.DCOSException))
```
Comment (NimaVaziri) [inline testing/sdk_install.py]: 👍

**Comment by elezar** [comment]
I'm pretty sure I added those imports! Thanks for picking them up...

**Comment by NimaVaziri** [comment]
@elezar seems like C* failures are fixed with this PR, should be able to merge once conflicts are resolved

**Comment by elezar** [comment]
@NimaVaziri if the tests pass, then I'm going to merge. The kafka error seems to be addressed in #1372.

**Comment by elezar** [comment]
@NimaVaziri, I reverted the changes to the hdfs tests (they failed, and I would rather address them properly in a separate PR).

**Comment by NimaVaziri** [comment]
Fair enough, I have another (hopefully final) fix for that error...

**Pull request merged**